src/utils/upload.ts

```
@@ -0,0 +1,41 @@
+import { IAllthingsRestClient } from '..'
+import { IFile } from '../rest/methods/file'
+
+export const upload = async (
+  attachments: ReadonlyArray<{
+    readonly content: Buffer
+    readonly filename: string
+  }>,
+  apiClient: IAllthingsRestClient,
+): Promise<ReadonlyArray<IFile | Error>> => {
+  const responses = await Promise.all(
+    attachments.map(async attachment => {
+      try {
+        const result = await apiClient.fileCreate({
+          file: attachment.content,
+          name: attachment.filename,
+        })
+
+        return result
+      } catch (error) {
+        return error
+      }
+    }),
+  )
+
+  return responses
+}
+
+export async function uploadFiles(
+  files: ReadonlyArray<{
+    readonly content: Buffer
+    readonly filename: string
+  }>,
+  client: IAllthingsRestClient,
+): Promise<ReadonlyArray<string>> {
+  const result = await upload(files, client)
+
+  return result
+    .filter((item): item is IFile => !(item instanceof Error))
```

is it ok just to filter out errors?

---

@yannvgn in my case it's good to filter them as a ticket should still be created and not fail if an attachment doesn't get uploaded, it should also add any successful uploads to the ticket. I've renamed this function and added a comment to make it more clear

---

Can you then not swallow the error but instead return a result that has either an object or an error like this:
```
{
  data?: item,
  error?: error,
}
```

---

@xeoxx I've updated this method to return both the successful and errors as suggested